plugins/inputs/logfile/tail/tail_queue.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tail
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/fdlimit"
+)
+
+/*
+	Centralize all the tailers which are scheduled for each file
+*/
+
+var defaultOpenFileLimit int = 1024
+
+type TailerEnqueue struct {
+	Queue chan bool
+}
+
+func NewTailerFifoQueue() *TailerEnqueue {
+	if allowedOpenFileLimit, err := fdlimit.CurrentOpenFileLimit(); err == nil {
+		defaultOpenFileLimit = allowedOpenFileLimit
```

Comment:
Declaring a "default" value and then updating the default value is bad code smell. What is the purpose of having this default to begin with? If there is a failure with deriving the allowed file descriptors for a given host, we're just going to assume it is 1024? What is our experience debugging issues like this going to be?

Reply:
The purpose of having the default would be avoid `interrupting the onboarding CloudWatchAgent for customers`. I'm indecisive about these two: would we `log.panic` if we cannot get file descriptors due to OS issue or should we allow the use the 1024 (the general upper bound for CWAgent which is used by customers). I'm still indecisive regarding about that so open to suggestion here.